NDTensors/ext/NDTensorscuTENSORExt/contract.jl

```diff
@@ -0,0 +1,41 @@
+using NDTensors: NDTensors, Tensor
+using NDTensors.Expose: unexpose
+using CUDA: CuArray
+using cuTENSOR: cuTENSOR
+function NDTensors.contract(
+  Etensor1::Exposed{<:CuArray},
+  labelstensor1,
+  Etensor2::Exposed{<:CuArray},
+  labelstensor2,
+  labelsoutput_tensor,
+)
+  tensor1 = unexpose(Etensor1)
+  tensor2 = unexpose(Etensor2)
+  ## reshape the flat data from ITensors into the correct dimensions
+  dA = reshape(data(tensor1), dims(tensor1))
+  dB = reshape(data(tensor2), dims(tensor2))
+  ## convert the ITensors into CuTensors
+  cutensorA = CuTensor(dA, [i for i in labelstensor1])
+  cutensorB = CuTensor(dB, [i for i in labelstensor2])
+
+  ## contract the CuTensors
+  cutensorC = cutensorA * cutensorB
+
+  ## TODO this is a first draft to this idea to see if the 
+  ## conversion works 
+  output_tensor = NDTensors.contraction_output(
```

Comment:
This isn't probably the best way to handle this especially since the next line I replace the storage/data with the data from the cutensor. But as a first draft, this does currently work and give numerically the same answers (from my small tests using dense)